### ISSUE TITLE
Vulnerability: Header Session Data Leak

### ISSUE THREAD

**Issue by evanshortiss** [evanshortiss/express-expeditious#2]
Currently it is possible to leak sensitive data when using this module. Here's an overview.

### Sample Code

```js
var expeditious = require('expeditious');
var expressExpeditious = require('express-expeditious');
var expeditiousInstance = expeditious({
  defaultTtl: (60 * 1000),
  engine: require('expeditious-engine-memory')()
  objectMode: true
});

// Our express application
var app = require('express')();

// Perform a login and return data plus set a cookie/session
app.post('/login', require('body-parser').json(), require('lib/routes/login'));

// Only users passing a valid session can get through this layer
app.use(require('lib/session'));

// Cache all responses to all routes by injecting expeditious first
app.use(expressExpeditious({
  expeditious: expeditiousInstance
}));

app.get('/data', (req, res) => {
  res.json(({
    'some': 'data'
  }));
})
```

### Explanation
In the example above, assume a user logs in, then they request `/data`. Assuming this request is the first, it will have the *entire* response generated stored in our cache including any updated cookie/session info we might send back as a header. This effectively means that a session could be leaked to another user.

I had not noticed this the applications I'd used this with are based on JWT tokens.

**Comment by evanshortiss** [comment]
If you wish to workaround this in the short term please implement a custom caching key function similar to the one [here](https://github.com/evanshortiss/express-expeditious#cache-key-generation-gencachekey)

**Comment by evanshortiss** [comment]
Fixed in 2.1.3
